packages/core-data/src/queried-data/selectors.js

```diff
@@ -55,7 +55,7 @@ function getQueriedItemsUncached( state, query ) {
 
 		// Having a target item ID doesn't guarantee that this object has been queried.
 		if ( ! state.items[ context ]?.hasOwnProperty( itemId ) ) {
-			return null;
+			continue;
```

Comment:
I'm not sure about this change, I wonder if this was there to ensure that we only return values if all the ids of the given query have been fetched.

Reply:
It doesn't seem like a bad change to me and we do the same with ids in `include`, but I might be missing some context. Maybe @jsnajdr or @Mamaduka have some input? 🤔

Reply:
I guess my question is: is this change required for this PR. If so, can you explain why/how?

Reply:
I think I tried to make a similar change in #34034 but eventually reverted it. Unfortunately, I don't have any additional input besides what was shared in that PR.

The lack of higher-level/integration tests here makes it easier for regressions to slip in even when changes are minor.

Reply:
>I guess my question is: is this change required for this PR. If so, can you explain why/how?

In this PR if we have fewer items received, we don't add the previous ones(that was the bug), but instead fill it with `undefined`. Without this change the query will not return anything.  Something I was thinking about was, wether to explicitly check for `undefined` and continue only then.

Reply:
What is the reason for filling with `undefined` instead of just removing them from the itemIds?

Reply:
Because we need the array(itemIds) to be sparse-like with `undefined` entries where holes exist, for pagination to work properly. If we don't do that and just remove the item there, my understanding is that can break other queries with different `page`.

Reply:
So I'm still not sure I understand the real reason, what things would break but regardless it seems safer to only "continue" here if the itemId is undefined like you suggested to minimize the impact of this PR.